### ISSUE TITLE
node.js/JS API

### ISSUE THREAD

**Issue by ugin** [mozilla/mentat#491]
I've looked over the wiki pages, on NPM and googled for a while but I can't find a way to use this in JS. I've noticed there are a few mentions about a JS API in older closed issues.

Will this project be maintained in the year? I would like to try it out for a project I'm currently developing.

**Comment by rnewman** [comment]
We're not in an active development cycle on Mentat; we're in the process of a broader analysis of the problem space as it relates to Firefox, including syncing.

I do expect us to return to this codebase; it's not dead. However, I will point out that Mentat isn't currently part of a shipping Mozilla product, so it has much less commitment than a project like NSS or Stylo, and it might change direction or be abandoned in the future with little notice.

We'd happily take PRs for any fixes or improvements you wanted to make, subject of course to the usual review process.

We don't have a JS interface for the current implementation. Mentat is implemented in Rust, so you'll want [Neon](https://github.com/neon-bindings/neon) to access it from Node. @bgrins made a first stab at this [here](https://github.com/bgrins/neon-mentat/); I expect you could continue that work.

Can you share any details about the project you're developing, @ugin?

**Comment by ugin** [comment]
I'm working on a CMS based on JavaScript and needed a _persistent_ data storage similar to Datomic. The focus of the project is to have a platform on which we can build webapps that look like websites, like a modern alternative to WordPress.

Initially I developed a data layer based on Datascript but then I realized I needed something external of the Node.js process as it was not feasible to persist the data on process crash.

I'm new to Rust, this week I planned to look over mentat and see if I can understand the source and hack on it.

Thank you for feedback
